cmd/prom-migrator/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/timescale/promscale/pkg/log"
+	plan "github.com/timescale/promscale/pkg/migration-tool/planner"
+	"github.com/timescale/promscale/pkg/migration-tool/reader"
+	"github.com/timescale/promscale/pkg/migration-tool/writer"
+)
+
+type config struct {
+	name               string
+	mint               int64
+	maxt               int64
+	readURL            string
+	writeURL           string
+	writerReadURL      string
+	progressEnabled    bool
+	progressMetricName string
+}
+
+func main() {
+	conf := new(config)
+	parseFlags(conf, os.Args[1:])
+	if err := log.Init(log.Config{Format: "logfmt", Level: "debug"}); err != nil {
+		_, _ = fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	log.Info("msg", fmt.Sprintf("%v+", conf))
+	if err := validateConf(conf); err != nil {
+		log.Error("msg", "could not parse flags", "error", err)
+		os.Exit(1)
+	}
+
+	planner, proceed, err := plan.CreatePlan(conf.mint, conf.maxt, conf.progressMetricName, conf.name, conf.writerReadURL, conf.progressEnabled, false)
```

Comment:
why not pass the config struct as an argument?

Reply:
Yeah, we can. Thank you for pointing out. Will implement in next commits.